reconcile/terraform_tgw_attachments.py

```diff
@@ -1,157 +1,213 @@
+        ocm = ocm_map.get(cluster_info["name"])
+        for peer_connection in cluster_info["peering"]["connections"]:
+            if peer_connection["provider"] == TGW_CONNECTION_PROVIDER:
+                yield from _build_desired_state_tgw_connection(
+                    peer_connection, cluster_info, ocm, awsapi
                 )
-                requester = {
-                    "tgw_id": tgw_id,
-                    "tgw_arn": tgw["tgw_arn"],
-                    "region": tgw["region"],
-                    "routes": tgw.get("routes"),
-                    "rules": tgw.get("rules"),
-                    "hostedzones": tgw.get("hostedzones"),
-                    "cidr_block": peer_connection.get("cidrBlock"),
-                    "account": account,
-                }
-                item = {
-                    "connection_provider": peer_connection_provider,
-                    "connection_name": connection_name,
-                    "requester": requester,
-                    "accepter": accepter,
-                    "deleted": peer_connection.get("delete", False),
-                }
-                desired_state.append(item)
 
-    return desired_state, error
 
+def _build_desired_state_tgw_connection(
+    peer_connection: dict,
+    cluster_info: dict,
+    ocm: OCM,
+    awsapi: AWSApi,
+) -> Generator[Optional[dict], Any, None]:
+    cluster_name = cluster_info["name"]
+    cluster_region = cluster_info["spec"]["region"]
+    cluster_cidr_block = cluster_info["network"]["vpc"]
 
-@defer
-def run(
-    dry_run, print_to_file=None, enable_deletion=False, thread_pool_size=10, defer=None
+    account = _enrich_account(
+        peer_connection, cluster_name, cluster_region, cluster_cidr_block, ocm
+    )
+
+    # accepter is the cluster's AWS account
+    accepter = _build_accepter(
+        peer_connection,
+        account,
+        cluster_region,
+        cluster_cidr_block,
+        awsapi,
+    )
+    if accepter["vpc_id"] is None:
+        logging.error(f"[{cluster_name}] could not find VPC ID for cluster")
+        yield None
+
+    account_tgws = awsapi.get_tgws_details(
+        account,
+        cluster_region,
+        cluster_cidr_block,
+        tags=json.loads(peer_connection.get("tags") or "{}"),
+        route_tables=peer_connection.get("manageRoutes"),
+        security_groups=peer_connection.get("manageSecurityGroups"),
+        route53_associations=peer_connection.get("manageRoute53Associations"),
+    )
+    for tgw in account_tgws:
+        connection_name = f"{peer_connection['name']}_{account['name']}-{tgw['tgw_id']}"
+        requester = _build_requester(peer_connection, account, tgw)
+        item = {
+            "connection_provider": TGW_CONNECTION_PROVIDER,
+            "connection_name": connection_name,
+            "requester": requester,
+            "accepter": accepter,
+            "deleted": peer_connection.get("delete", False),
+        }
+        yield item
+
+
+def _enrich_account(
```

Comment:
enrich is pretty generic name, could you be more specific?

Reply:
something related to `assume`?

Reply:
Yeah, perhaps `_set_aws_assume_role_data` ?

Reply:
updated, how about `_account_with_assume_role_data`? want to indicate the return is an account